### PR TITLE
Tree

### PR DESCRIPTION
**Pull request by wosunkwo**

### CODE REVIEW

**Comment by wosunkwo** [comment]
- [x] Create a Binary Tree Class
- [x] Create a Binary Search Tree Class
- [x] Create test case for Binary Tree Class
- [x] Create test case for Binary Search Tree Class

File: code401-challenges/src/main/java/code401/challenges/tree/BinaryTree.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    }
+
+    public ArrayList<T> postOrder(Node<T> node){
+        ArrayList<T> resultArr = new ArrayList<>();
+        postOrderHelper(node, resultArr);
+        return resultArr;
+    }
+
+    public void preOrderHelper(Node<T> node, ArrayList<T> resultArr){
+        try{
```
Comment (evanslaton) [inline code401-challenges/src/main/java/code401/challenges/tree/BinaryTree.java:44]: I'm not sure these try/catches are necessary

File: code401-challenges/src/test/java/code401/challenges/tree/BinarySearchTreeTest.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        classUnderTest = new BinarySearchTree();
+    }
+
+    @Test
+    public void testInstantiateEmptyTree(){
+        assertNull("This should return true if the tree is empty/null ", classUnderTest.getRoot());
+    }
+
+    @Test
+    public void testSingleRootNodeTree(){
```
Comment (evanslaton) [inline code401-challenges/src/test/java/code401/challenges/tree/BinarySearchTreeTest.java:22]: This root doesn't actually test if the tree contains one node, it checks to see if the tree contains a node that has a value of 1. To check if this tree has one node you could have checked that the `root` wasn't `null` and that that the `root`s left and right were also `null`

File: code401-challenges/src/test/java/code401/challenges/tree/BinarySearchTreeTest.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        classUnderTest.add(1);
+        assertTrue("This should return true if the tree contains a node ",classUnderTest.contains(1));
+    }
+
+    @Test
+    public void testLeftRightRootNodeTree(){
```
Comment (evanslaton) [inline code401-challenges/src/test/java/code401/challenges/tree/BinarySearchTreeTest.java:28]: This doesn't actually check if the node were correctly inserted into the left and right of the root node

**Pull request merged**